### PR TITLE
Companion PR for 'Simplify `NativeExecutionDispatch` and remove the `native_executor_instance!`'  (9562)

### PR DESCRIPTION
**Pull request by expenses**
https://github.com/paritytech/substrate/pull/9562

### CODE REVIEW

**Review by gilescope — APPROVED** [review]

File: node/client/src/lib.rs [hunk 1]
```diff
@@ -39,36 +38,67 @@ pub type FullBackend = sc_service::TFullBackend<Block>;
 
 pub type FullClient<RuntimeApi, Executor> = sc_service::TFullClient<Block, RuntimeApi, Executor>;
 
-native_executor_instance!(
-	pub PolkadotExecutor,
-	polkadot_runtime::api::dispatch,
-	polkadot_runtime::native_version,
-	frame_benchmarking::benchmarking::HostFunctions,
-);
+pub struct PolkadotExecutor;
```
Comment (kianenigma) [inline node/client/src/lib.rs:41]: pub items not documented.

**Review by kianenigma — APPROVED** [review]
LGTM

File: node/test/service/src/lib.rs [hunk 2]
```diff
@@ -55,12 +54,21 @@ use substrate_test_client::{
...
 };
 
-native_executor_instance!(
-	pub PolkadotTestExecutor,
-	polkadot_test_runtime::api::dispatch,
-	polkadot_test_runtime::native_version,
-	frame_benchmarking::benchmarking::HostFunctions,
-);
+// Declare an instance of the native executor named `Executor`. Include the wasm binary as the
+// equivalent wasm code.
```
Comment (expenses) [inline node/test/service/src/lib.rs:58]:
```suggestion
/// Declare an instance of the native executor named `Executor`. Include the wasm binary as the
/// equivalent wasm code.
```

**Review by drahnr — APPROVED** [review]
LGTM

**Review by HCastano — APPROVED** [review]
Bridges code looks fine 👍

**Pull request merged**